pandas/tests/generic/test_frame.py

```diff
@@ -268,3 +268,14 @@ def test_deepcopy_empty(self):
         empty_frame_copy = deepcopy(empty_frame)
 
         self._compare(empty_frame_copy, empty_frame)
+
+    def test_nested_dict_construction(self):
+        columns = ['Nevada', 'Ohio']
+        pop = {'Nevada': {2001: 2.4, 2002: 2.9},
+               'Ohio': {2000: 1.5, 2001: 1.7, 2002: 3.6}}
+        result = pd.DataFrame(pop, index=[2001, 2002, 2003], columns=columns)
+        expected = pd.DataFrame(
+            {'Nevada': [2.4, 2.9, np.nan], 'Ohio': [1.7, 3.6, np.nan]},
```

Comment:
Maybe better to construct this with a list of lists? Would put an extra safeguard in place in case of something wonky going on with dict construction

Reply:
Done